src/qt/receivecoinsdialog.cpp

```diff
@@ -70,15 +70,17 @@ void ReceiveCoinsDialog::setModel(WalletModel *_model)
 
     if(_model && _model->getOptionsModel())
     {
-        _model->getRecentRequestsTableModel()->sort(RecentRequestsTableModel::Date, Qt::DescendingOrder);
+        assert(m_recent_requests_table_model == nullptr);
+        m_recent_requests_table_model = new RecentRequestsTableModel(_model);
```

Comment:
No `delete` necessary in this case? I tend to prefer it occur even if not strictly necessary.

Reply:
Delete on `setModel(nullptr)`?

Reply:
> No `delete` necessary in this case?

`m_recent_requests_table_model` is a `QObject` with a parent. This ensures that it will be deleted during its parent deletion, no?

Reply:
Yes. Only thing "wrong" is that when `WalletModel` is destroyed we will have a dangling
`m_recent_requests_table_model`. But AFAICT this shouldn't be a concert because that happens when the wallet is being unloaded and everything is being destroyed.

Anyway, I'm thinking we could move away from these `setModel(WalletModel*)` and pass it on the constructor, but it's going to take a while to get there.

Reply:
> Only thing "wrong" is that when `WalletModel` is destroyed we will have a dangling
> `m_recent_requests_table_model`.

Sounds not good...

>  But AFAICT this shouldn't be a concern because that happens when the wallet is being unloaded and everything is being destroyed.

Agree.

> Anyway, I'm thinking we could move away from these `setModel(WalletModel*)` and pass it on the constructor, but it's going to take a while to get there.

Maybe discuss this on IRC meeting?

Reply:
> Maybe discuss this on IRC meeting?

Not sure worth it, a draft PR with the whole change might be better.

Reply:
This will create a new  `RecentRequestsTableModel` every time `setModel` is called. I do not think this is expected behavior. `setModel` should set a model not create one.

Reply:
This is called once, see the assertion above.

There is no use case for switching models, that's why I think we can slowly refactor to get the dependencies on the constructor rather than having these setters, which add unnecessary state management.